osu.Framework/Utils/OrderAttribute.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace osu.Framework.Utils
+{
+    public static class OrderAttributeUtils
+    {
+        /// <summary>
+        /// Get values of an enum in order. Supports custom ordering via <see cref="OrderAttribute"/>.
+        /// </summary>
+        public static IEnumerable<T> GetValuesInOrder<T>()
+        {
+            var type = typeof(T);
+
+            if (!type.IsEnum)
+                throw new InvalidOperationException("T must be an enum");
```

Comment:
I know it was like this game-side, but it was arguably unhelpful/wrong - comments to this file are just fixes that we might as well go ahead with while we're moving things around

```suggestion
                throw new InvalidOperationException($"{typeof(T)} must be an enum");
```